google-cloud-spanner/src/test/java/com/google/cloud/spanner/SessionPoolTest.java

```diff
@@ -1628,7 +1628,7 @@ public Void call() {
     assertThat(record.getMetrics())
         .containsEntry(MetricRegistryConstants.NUM_ACQUIRED_SESSIONS, 0L);
     assertThat(record.getMetrics())
-        .containsEntry(MetricRegistryConstants.NUM_RELEASED_SESSIONS, 3L);
+        .containsEntry(MetricRegistryConstants.NUM_RELEASED_SESSIONS, 5L);
```

Comment:
@skuruppu thanks for the suggestions. I made the changes, but somehow test results look fishy to me. After change, NUM_RELEASED_SESSIONS == 5 and NUM_ACQUIRED_SESSIONS == 0.

Reply:
I think I was partially wrong. It turns out that `releaseSession()` is only used internally. So you can get calls to this function due to the session pool doing keep alive requests (e.g. check out session to send a "select 1" then release it back to the pool). I think this is why you're seeing the number of released sessions being higher than you expected.

I also just realized that `getReadSession()` docs say to call `Session.close()` when you're done with the sessions. It seems that sessions acquired by the user will get closed and sessions acquired by the client library gets released.

@olavloite did I understand this correctly?

Reply:
I don't understand why `NUM_ACQUIRED_SESSIONS` is zero though. I would've thought calling `getReadSession()` would increment this.

Reply:
@olavloite would like to get your eyes on this.